js/script.js

```diff
@@ -0,0 +1,111 @@
+var Click = (function() {
+  this.numClicks = 0;
+  this.arr = [];
+});
+
+var clickTd = new Click();
+
+//Start With click
+$('body').on('click', 'td', (function(e) {
+  var _this = this;
+  var tdClass = $(_this).children().attr('class');
+  var child = $(_this).children();
+  $(child).show();
+  setTimeout(function() {
+    clickTd.incrementClicks();
+    clickTd.addToArr(tdClass);
+  }, 500);
+}));
```

Comment:
I'd suggest putting the code that uses a class below all of the code that sets up the class (e.g. adds the methods to it) – otherwise you can get into a weird situation where that class might get instantiated, but the methods aren't available on it yet.